src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -215,36 +304,47 @@ def add(self, mapping: PathMapping) -> None:
         """
         self._add_mapping(mapping.src, mapping.dest)
 
-    def _mappings_for_source(
+    def _expand_artifact_mapping(
         self,
         src: Path,
+        dest: Path,
         absolute: bool = False,
         expand_directories: bool = False,
         predicate: ArtifactPredicate = lambda src, dest: True,
     ) -> Iterator[Tuple[Path, Path]]:
+        """
+        Expands the specified source-destination mapping according to the provided options.
+        The original mapping is yielded, followed by any expanded mappings derived from
+        it.
+
+        Arguments:
+            src {Path} -- the source path
+            dest {Path} -- the destination path
+            absolute {bool} -- when True, all mappings will be yielded as absolute paths
+            expand_directories {bool} -- when True, child mappings are yielded if the source path is a directory.
+            predicate {ArtifactPredicate} -- when specified, only mappings satisfying this predicate will be yielded.
+        """
         canonical_src = self._canonical_src(src)
-        canonical_dests = self._src_to_dest.get(canonical_src)
-        assert canonical_dests is not None
+        canonical_dest = self._canonical_dest(dest)
```

Comment:
Also unclear to me. Are we losing information here by only getting the one `canonical_dest` instead of the list of the values like in the original map?

Reply:
No, before the logic was:

```
for src in all sources:
   for dest in all src -> dest mappings:
        <process>
```

but now it's

```
for src -> dest in all mappings:
     <process>
```

it's really the same thing, I just moved the iteration logic around a little.